packages/rest-api/src/controllers/swapTxInfoController.ts

```diff
@@ -10,23 +12,34 @@ export const swapTxInfoController = async (req, res) => {
   }
 
   try {
-    const { chain, amount } = req.query
-    const fromTokenInfo = res.locals.tokenInfo.fromToken
-    const toTokenInfo = res.locals.tokenInfo.toToken
+    const { chain, amount, address, fromToken, toToken } = req.query
+
+    if (!isAddress(fromToken) || !isAddress(toToken)) {
+      return res.status(400).json({ error: 'Invalid token address' })
+    }
+
+    const fromTokenInfo = tokenAddressToToken(chain.toString(), fromToken)
+    const toTokenInfo = tokenAddressToToken(chain.toString(), toToken)
+
+    if (!fromTokenInfo || !toTokenInfo) {
+      return res
+        .status(400)
+        .json({ error: 'Token not supported on specified chain' })
+    }
 
     const amountInWei = parseUnits(amount.toString(), fromTokenInfo.decimals)
 
     const quote = await Synapse.swapQuote(
       Number(chain),
-      fromTokenInfo.address,
-      toTokenInfo.address,
+      fromToken,
+      toToken,
       amountInWei
     )
 
     const txInfo = await Synapse.swap(
       Number(chain),
-      fromTokenInfo.address,
-      toTokenInfo.address,
+      address,
+      toToken,
```

Comment:
This has to be `fromToken`, the swap function takes the initial token/amount as a parameter (while the result token/min amountOut /deadline are provided in the query)